src/test/java/org/opensearch/knn/index/query/KNNQueryBuilderTests.java

```diff
@@ -127,6 +127,43 @@ public void testFromXcontent_WithFilter() throws Exception {
         actualBuilder.equals(knnQueryBuilder);
     }
 
+    public void testFromXContent_invalidQueryVectorType() throws Exception {
+        final ClusterService clusterService = mockClusterService(Version.CURRENT);
+
+        final KNNClusterUtil knnClusterUtil = KNNClusterUtil.instance();
+        knnClusterUtil.initialize(clusterService);
+
+        String[] invalidTypeQueryVector = { "a", "b", "c", "d" };
+
+        XContentBuilder builder = XContentFactory.jsonBuilder();
+        builder.startObject();
+        builder.startObject(FIELD_NAME);
+        builder.field(KNNQueryBuilder.VECTOR_FIELD.getPreferredName(), invalidTypeQueryVector);
+        builder.field(KNNQueryBuilder.K_FIELD.getPreferredName(), K);
+        builder.endObject();
+        builder.endObject();
+        XContentParser contentParser = createParser(builder);
+        contentParser.nextToken();
+        expectThrows(IllegalArgumentException.class, () -> KNNQueryBuilder.fromXContent(contentParser));
```

Comment:
Can we verify the error message here as well?

Reply:
Added